python/amici/petab_objective.py

```diff
@@ -1,181 +1,534 @@
-"""Functionality related to evaluating the objective function for a PEtab
-problem"""
+"""Functionality related to running simulations or evaluating the objective
+function as defined by a PEtab problem"""
 
-import numbers
-import numpy as np
-import pandas as pd
 import copy
+import logging
+import numbers
+from typing import List, Sequence, Optional, Dict, Tuple, Union, Any
 
-from typing import List, Sequence
 import amici
+import numpy as np
+import pandas as pd
 import petab
+from amici.logging import get_logger
+from petab.C import *
+
+
+logger = get_logger(__name__)
+
+
+def simulate_petab(petab_problem: petab.Problem,
```

Comment:
what is the purpose of this function? for debugging/testing? since the edatas are created each time, it would be rather inefficient as an objective function.

Reply:
It's used for the benchmark model test cases and it's quite helpful for debugging/testing, because it avoids lot's of boilerplate code. Not meant to be used for most efficient parameter estimation. Although I guess it will do the job, if not working with large amounts of data.